resource/packages/rpm/packageManager.go

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpm
+
+import (
+	"fmt"
+	"os/exec"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+// PackageVersion is a type alias for a string, since various packages may use
+// different naming conventions
+type PackageVersion string
+
+// PackageManager describes an interface for managing packages and helps make
+// `Check` and `Apply` testable.
+type PackageManager interface {
```

Comment:
some of the interfaces and functions in this file looks like they could be applicable to the apt resource also